src/pages/blog.tsx

```diff
@@ -1,34 +1,91 @@
 import React from 'react';
+import styled from 'styled-components';
+import { graphql, useStaticQuery } from 'gatsby';
+import { FluidObject } from 'gatsby-image';
 
-import Layout from '../components/layout/layout';
 import SEO from '../components/seo';
-import { PageTitle } from '../components/typography/typography';
+import Layout from '../components/layout/layout';
+import { up } from '../components/breakpoint/breakpoint';
 import { Grid, GridItem } from '../components/grid/grid';
-import { graphql, Link, useStaticQuery } from 'gatsby';
+import { PageTitle } from '../components/typography/typography';
+import { BlogCard } from '../components/cards/blog-card';
+
+const TOP_POST_COUNT = 2;
+
+const LOCALE = 'de-DE';
+const DATE_OPTIONS = {
+  year: 'numeric',
+  month: 'long',
+  day: 'numeric',
+};
 
 interface AllBlogPostsQuery {
   allMarkdownRemark: {
     nodes: {
       id: string;
+      excerpt: string;
       frontmatter: {
         date: string;
         path: string;
         title: string;
+        previewImage?: boolean;
       };
+      rawMarkdownBody: string;
     }[];
   };
+  blogPlaceholderImage: {
+    childImageSharp: {
+      fluid: FluidObject;
+    };
+  };
 }
 
+const BlogPageTitle = styled(PageTitle)`
+  margin-bottom: 48px;
+
+  ${up('md')} {
+    margin-bottom: 80px;
+  }
+`;
+
+const getDateString = (date: string): string => {
```

Comment:
This function is quite complicated to understand, here are some tips:

1. Dates are always a hassle. Especially if you also consider that different browsers treat them differently. Therefore, using a date library like `date-fns` is very much appreciated. They have simple and easy to read helper methods. So I would suggest you add this library to this project and use those. Example:
1.1 You could replace the whole method with https://date-fns.org/v2.14.0/docs/formatRelative (not sure if that does exactly what we need)
1.2 If the `formatRelative` doesn't work/ fit, you can replace the calculation of `isToday` and `isYesterday` with https://date-fns.org/v2.14.0/docs/isYesterday and https://date-fns.org/v2.14.0/docs/isToday.

2. The naming of "getDateString" is quite general. Maybe a more specific name like "formatToRelative" is more clear.

3. The `date` parameter is of type string. As a date can have a bunch of different string representations (ISO or custom format like `dd-mm-yyyy`) it should be made clear in a comment above what format is expected.

Reply:
1. I disagree. To pull in a library to handle one date is overkill, when this function achieves the same result.
2. Sure, I'll rename the function, but `formatToRelative` is not entirely correct, because the method not only determines "Gestern" & "Heute", which wasn't even asked for. But also parses the date to an understandable, nicely readable date format. In detail yyyy-mm-dd is parsed to 01. August 2020, as the input date format, can be confusing (US/BE).
3. Okay.

Reply:
_Edit_: I simplified the method, as in my honest opinion, as this was somewhat of a suggestion & not required and this isn't of importance at the moment. 🙂

Reply:
Yeah, there should always be valid reasons to import a library. For dates, most of the use cases are a valid reason to import one. Even with the simple method here, `date-fns` parse method would be better. Why? Because using a string to initiate a date is highly discourage (see "3. Timestamp string" of the spec: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/Date). 

Technically, it's not an overkill. As date-fns uses ES6 modules, only the functions that are imported will be used, all other code of the library is untouched. Also test coverage and documentation will probably most of the time better with such (let's call it) "high quality libraries". 

But as for our simple usecase everything is working fine, so I'm good to go 👍